### DISCUSSION TITLE
Mass-delete identical points from map view

### DISCUSSION THREAD

**Discussion by ajpanton** [Freika/dawarich#433] [Ideas]
In the map view, clicking on a point brings a window with a "Delete" button. I'd like to also have a "Delete all" button that deletes all points that have identical coordinates (within the visible time range).

Some points are sourced from Wifi or cellular, and those can be very inaccurate at times. Often there are some very specific discrete locations that are associated with these sources, so I might have tens or hundreds of points with identical coordinates. If one of these locations is so wildly wrong that I want to remove all instances of it, it's a lot of work deleting them all one by one. I could look those up in the points tab, but that's also of limited use as one can only filter points by time.

A "Delete all" would help in this case, deleting all instances of a specific location. To be safe, it should only delete points that are in the currently visible time range.

**Comment by nareddyt** [comment]
Hi @Freika , I would like to contribute this feature.

It seems fairly straightforward, we'd essentially add a `Delete All` button to the point. The button would end up running the following SQL query:

```
DELETE FROM POINTS
WHERE lonlat = '<LONLAT FROM SELECTED POINT>';
```

WDYT? Has there been any other asks for this feature, and would you recommend implementing it via a different interface other than a `Delete All` button on each point?

**Reply by Freika** [reply]
Hi, thanks for suggestion! I don't think there is a big demand for this feature. For power users it's probably not an issue to spin up the console or psql and execute a request such as the one you provided or basically any other with the help of chatgpt

I also don't want to bloat interface with something that was requested only once

**Reply by nareddyt** [reply]
Ok fair enough, thanks for the feedback

**Reply by goinsandrew** [reply]
I personally would love this feature. I recently imported my google timeline data for the last 14(!) years, but it lists numerous instances of me being in foriegn countries. after a bit of pondering, I realized most of those are from the post-covid period where I was geo spoofing in mobile games. Ideally, I would actually see this delete all button added to the select area tool. As it is, I am currently setting aside 30 minutes a day to gradually pluck out all these points....

**Reply by Freika** [reply]
It's gonna be a thing in one of the following releases, via selection tool: select an area with points, hit delete, hit confirm and they're gone

**Reply by Freika** [reply]
Done in 0.35.0
